net/libfab/libfab.c

```diff
@@ -770,7 +774,7 @@ static uint32_t libfab_handle_connect_request_events(struct m0_fab__tm *tm)
 	if (rc >= (int)sizeof(struct fi_eq_cm_entry) && event == FI_CONNREQ) {
 		cm_entry = (struct fi_eq_cm_entry *)entry;
 		cd = (struct m0_fab__conn_data*)(cm_entry->data);
-		libfab_straddr_gen(cd, straddr, sizeof(straddr), &en);
+		libfab_straddr_gen(cd, straddr, ARRAY_SIZE(straddr), &en);
```

Comment:
because the element size of `char` is one byte, so the `sizeof` has the same result of 'ARRAY_SIZE`.

But usually, we use sizeof to get the size of its memory consumption.

Reply:
String symbols can be several bytes, e.g. wchar_t... So it's always better to use ARRAY_SIZE() for strings, than sizeof().

Reply:
Yes, you are right.